nlu.py

```diff
@@ -20,6 +20,34 @@ def _get_commands(text_sequence: [str]) -> set:
     return possible_commands
 
 
+def _consine_similarity(question: [str], text_sequence: [str]) -> float:
+    words = set(question+text_sequence)
+    question_frequency = _get_frequency(words, question)
+    text_frequency = _get_frequency(words, text_sequence)
```

Comment:
Seems would have a bug? Your Ai and Bi could be wrong here, see https://stackoverflow.com/questions/1867861/dictionaries-how-to-keep-keys-values-in-same-order-as-declared

You should use an ordered map type, or better, try counting frequency at here.

Reply:
but the get_frequency method return list type, it still wrong?